test/e2e/resize_nodes.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	// compute "code.google.com/p/google-api-go-client/compute/v1"
+	// "google.golang.org/cloud/compute/metadata"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/wait"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func resizeNodeInstanceGroup(size int) error {
+	// TODO: make this hit the compute API directly instread of shelling out to gcloud.
+	output, err := exec.Command("gcloud", "preview", "managed-instance-groups", "--zone="+testContext.CloudConfig.Zone,
+		"resize", testContext.CloudConfig.NodeInstanceGroup, fmt.Sprintf("--new-size=%v", size)).CombinedOutput()
+	if err != nil {
+		Logf("Failed to resize node instance group: %v", string(output))
+	}
+	return err
+}
+
+func waitForNodeInstanceGroupSize(size int) error {
+	retryCount := 12
+	retryInterval := 20 * time.Second
+	for i := 0; i < retryCount; i++ {
```

Comment:
Can you please switch this to use the timing mechanism (for loop with explicit timeout) we're using in the e2e tests, such as:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/util.go#L93